### ISSUE TITLE
what happened to container.RegisterTypes

### ISSUE THREAD

**Issue by vip32** [unitycontainer/unity#143]
in 4.0 we could use the following:
```
container.RegisterTypes(AllClasses.FromAssembliesInBasePath()
                .Where(type => typeof(IInterface).IsAssignableFrom(type)),
                   WithMappings.FromAllInterfaces,
                   WithName.TypeName,
                   WithLifetime.Transient);
```
however that does not compile anymore against 5.0.2. if this has been removed this release is not backward compatible, as it was supposed to be

**Comment by vip32** [comment]
this will probably help out
https://www.nuget.org/packages/Unity.RegistrationByConvention/

**Comment by vip32** [comment]
ok it seems unity 5.0.2 needs unity.abstractions 2.0.0
however Unity.RegistrationByConvention needs unity.abstractions 2.0.**2**

that's why the 2 cannot be used easily together now

**Comment by ENikS** [comment]
Still working out deployment kinks, sorry for the mix up. Will release consistent lineup this weekend.

**Comment by AmitBhatnagar24** [comment]
I too updated my project from 4.0.1 to 5.0.2 and ran into this snag. I couldn't find any updated samples or "migrating from v4 to v5" type documentation anywhere..

Can someone point me in the proper direction to understand what I need to do to continue to use Unity?

**Comment by ENikS** [comment]
There is no directions as of yet. We are all doing it for the first time, me included. 
You could log what you do and contribute directions for others to follow...

**Comment by AmitBhatnagar24** [comment]
I'm sure this is going to be upsetting to most who upgraded because nuget said there was a newer version available. No accompanying documentation to migrate to the new version(?)

**Comment by ENikS** [comment]
As I said, we just restarting this project and have no manpower to do it all. Would you like to contribute new documentation so others can follow?

**Comment by ENikS** [comment]
Fixed

**Comment by vip32** [comment]
Great to see unity getting some love after all those years @ENikS. If you need some help please call it out!

**Comment by ENikS** [comment]
Thank you for the offer. I have plenty of things to do if you have time and experience. 
For example .Net Core port of Unity.RegistrationByConvention still not done.

**Comment by vip32** [comment]
Unity.RegistrationByConvention still uses an outdated Unity.Abstractions :(

error CS1705: Assembly 'Unity.Container' with identity 'Unity.Container, Version=5.1.1.0, Culture=neutral, PublicKeyToken=489b6accfaf20ef0' uses 'Unity.Abstractions, Version=**2.1.1.0**, Culture=neutral, PublicKeyToken=6d32ff45e0ccc69f' which has a higher version than referenced assembly 'Unity.Abstractions' with identity 'Unity.Abstractions, Version=**2.1.0.0**, Culture=neutral, PublicKeyToken=6d32ff45e0ccc69f'

**Comment by vip32** [comment]
looking into your solution, how are these nugets build and published? as they are all in one sln i was expecting one build and multiple publishes (with same version) as they all have direct project references.

**Comment by ENikS** [comment]
Each repo has package.sln

**Comment by ENikS** [comment]
This is where these dependencies are specified: https://github.com/unitycontainer/container/blob/v5.x/package.props

**Comment by vip32** [comment]
i see a fresh unity.abstractions 2.1.**1** on nuget, let me try again
https://www.nuget.org/packages/Unity.Abstractions/

**Comment by AmitBhatnagar24** [comment]
Everything worked just fine for me after updating everything to the latest version.
